backend-lib/src/main/rust/lib.rs

```diff
@@ -1855,21 +1848,19 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_proposeTr
         let network = parse_network(network_id as u32)?;
         let mut db_data = wallet_db(env, network, db_data)?;
         let account_uuid = account_id_from_jni(&env, account_uuid)?;
-        let to = utils::java_string_to_rust(env, &to);
+        let to = utils::java_string_to_rust(env, &to)?;
         let value = Zatoshis::from_nonnegative_i64(value)
             .map_err(|_| anyhow!("Invalid amount, out of range"))?;
 
         let to = to
             .parse()
             .map_err(|e| anyhow!("Can't parse recipient address: {}", e))?;
 
-        let memo = if memo.is_null() {
-            None
-        } else {
-            MemoBytes::from_bytes(&env.convert_byte_array(memo)?)
-                .map(Some)
-                .map_err(|e| anyhow!("Invalid MemoBytes: {}", e))?
-        };
+        let memo = utils::java_nullable_bytes_to_rust(env, &memo)?
+            .as_deref()
+            .map(MemoBytes::from_bytes)
+            .transpose()
+            .map_err(|e| anyhow!("Invalid MemoBytes: {}", e))?;
```

Comment:
This pattern occurs twice (also in `RustBackend.proposeShielding`); consider adding a `nullable_memo_from_jni` helper.